src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_AMOUNT = 5;
+
+    public Figure getRandomFigure() {
+        ColorSupplier supplier = new ColorSupplier();
+        Random random = new Random();
+
+        Circle circle = new Circle(
+                "Circle",
+                random.nextInt(100),
+                supplier.getRandomColor()
+        );
+
+        IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(
+                "IsoscelesTrapezoid",
+                random.nextInt(100),
+                random.nextInt(100),
+                random.nextInt(100),
+                supplier.getRandomColor()
+        );
+
+        Rectangle rectangle = new Rectangle(
+                "Rectangle",
+                random.nextInt(100),
+                random.nextInt(100),
+                supplier.getRandomColor()
+        );
+
+        RightTriangle rightTriangle = new RightTriangle(
+                "RightTriangle",
+                random.nextInt(100),
+                random.nextInt(100),
+                supplier.getRandomColor()
+        );
+
+        Square square = new Square(
+                "Square",
+                random.nextInt(100),
+                supplier.getRandomColor()
+        );
```

Comment:
you're wasting the memory, use the switch case to create the figure only in case this one should be created

Reply:
You can create an enum of figure types and choose randomly one of type 
after that switch through this type and create object if it matches